Readme.md
# Bank Ru Payment (Безналичный платеж)

## About

Модуль для PrestaShop (престашоп) 1.6

Прием безналичных платежей, счет 



CHANGELOG.txt
2018-12-15 10:00:00 +0300	// Делаю


translations/ru.php
<?php

global $_MODULE;
$_MODULE = array();

$_MODULE['<{bankrupay}prestashop>bankrupay_05adcee99142c1a60fb38bb1330bbbc1'] = 'Банковский перевод';
$_MODULE['<{bankrupay}prestashop>bankrupay_a246a8e9907530c4c36b8b4c37bbc823'] = 'Приём оплаты товаров банковским переводом.';
$_MODULE['<{bankrupay}prestashop>bankrupay_cbe0a99684b145e77f3e14174ac212e3'] = 'Вы точно хотите удалить эти данные?';
$_MODULE['<{bankrupay}prestashop>bankrupay_0ea0227283d959415eda0cfa31d9f718'] = 'Для использования этого модуля нужно задать владельца счета и реквизиты.';
$_MODULE['<{bankrupay}prestashop>bankrupay_a02758d758e8bec77a33d7f392eb3f8a'] = 'Не задана валюта для этого модуля.';
$_MODULE['<{bankrupay}prestashop>bankrupay_bfa43217dfe8261ee7cb040339085677'] = 'Необходимо задать реквизиты.';
$_MODULE['<{bankrupay}prestashop>bankrupay_ccab155f173ac76f79eb192703f86b18'] = 'Требуется указать владельца счета.';
$_MODULE['<{bankrupay}prestashop>bankrupay_c888438d14855d7d96a2724ee9c306bd'] = 'Настройки обновлены';
$_MODULE['<{bankrupay}prestashop>bankrupay_4ffaad55a1d22c453e7c9bad67b0598f'] = 'Оплата банковским переводом';
$_MODULE['<{bankrupay}prestashop>bankrupay_5dd532f0a63d89c5af0243b74732f63c'] = 'Контактная информация';
$_MODULE['<{bankrupay}prestashop>bankrupay_857216dd1b374de9bf54068fcd78a8f3'] = 'Владелец счета';
$_MODULE['<{bankrupay}prestashop>bankrupay_3ec365dd533ddb7ef3d1c111186ce872'] = 'Подробности';
$_MODULE['<{bankrupay}prestashop>bankrupay_6b154cafbab54ba3a1e76a78c290c02a'] = 'Отделение банка, IBAN номер, БИК и т.д.';
$_MODULE['<{bankrupay}prestashop>bankrupay_f9a1a1bb716cbae0503d351ea2af4b34'] = 'Адрес банка';
$_MODULE['<{bankrupay}prestashop>bankrupay_c9cc8cce247e49bae79f15173ce97354'] = 'Сохранить';
$_MODULE['<{bankrupay}prestashop>validation_e2b7dec8fa4b498156dfee6e4c84b156'] = 'Метод оплаты не доступен.';
$_MODULE['<{bankrupay}prestashop>payment_execution_644818852b4dd8cf9da73543e30f045a'] = 'Вернуться к оформлению заказа';
$_MODULE['<{bankrupay}prestashop>payment_execution_6ff063fbc860a79759a7369ac32cee22'] = 'Оформление заказа';
$_MODULE['<{bankrupay}prestashop>payment_execution_511e8b930b4d3d6002984c0373eb6d4b'] = 'Оплата банковским переводом';
$_MODULE['<{bankrupay}prestashop>payment_execution_f1d3b424cd68795ecaa552883759aceb'] = 'Информация о заказе';
$_MODULE['<{bankrupay}prestashop>payment_execution_879f6b8877752685a966564d072f498f'] = 'Корзина пуста.';
$_MODULE['<{bankrupay}prestashop>payment_execution_05adcee99142c1a60fb38bb1330bbbc1'] = 'Банковский перевод';
$_MODULE['<{bankrupay}prestashop>payment_execution_afda466128ee0594745d9f8152699b74'] = 'Вы выбрали оплату банковским переводом.';
$_MODULE['<{bankrupay}prestashop>payment_execution_c884ed19483d45970c5bf23a681e2dd2'] = 'Краткая информация по вашему заказу:';
$_MODULE['<{bankrupay}prestashop>payment_execution_e2867a925cba382f1436d1834bb52a1c'] = 'Общая сумма вашего заказа';
$_MODULE['<{bankrupay}prestashop>payment_execution_1f87346a16cf80c372065de3c54c86d9'] = '(вкл. НДС)';
$_MODULE['<{bankrupay}prestashop>payment_execution_b28be4c423d93e02081f4e79fe2434e8'] = 'Мы принимаем валютные переводы.';
$_MODULE['<{bankrupay}prestashop>payment_execution_a7a08622ee5c8019b57354b99b7693b2'] = 'Выберите одно из следующих действий:';
$_MODULE['<{bankrupay}prestashop>payment_execution_a854d894458d66d92cabf0411c499ef4'] = 'Мы принимаем переводы в следующих валютах:';
$_MODULE['<{bankrupay}prestashop>payment_execution_3dd021316505c0204989f984246c6ff1'] = 'Информация о банковском счете будет показана на следующей странице.';
$_MODULE['<{bankrupay}prestashop>payment_execution_edd87c9059d88fea45c0bd6384ce92b9'] = 'Пожалуйста подтвердите свой заказ, нажав "Я подтверждаю ​​заказ".';
$_MODULE['<{bankrupay}prestashop>payment_execution_46b9e3665f187c739c55983f757ccda0'] = 'Я подтверждаю заказ';
$_MODULE['<{bankrupay}prestashop>payment_execution_569fd05bdafa1712c4f6be5b153b8418'] = 'Другие способы оплаты';
$_MODULE['<{bankrupay}prestashop>infos_c1be305030739396775edaca9813f77d'] = 'Этот модуль позволяет принимать платежи банковским переводом.';
$_MODULE['<{bankrupay}prestashop>infos_60742d06006fde3043c77e6549d71a99'] = 'Если клиент выбирает оплату банковским переводом, статус заказа изменится на "В ожидании оплаты."';
$_MODULE['<{bankrupay}prestashop>infos_5fb4bbf993c23848433caf58e6b2816d'] = 'Вы должны вручную подтвердить заказ по факту зачисления перевода.';
$_MODULE['<{bankrupay}prestashop>payment_return_88526efe38fd18179a127024aba8c1d7'] = 'Ваш заказ на %s оформлен.';
$_MODULE['<{bankrupay}prestashop>payment_return_1f8cdc30326f1f930b0c87b25fdac965'] = 'Реквизиты для банковского перевода:';
$_MODULE['<{bankrupay}prestashop>payment_return_b2f40690858b404ed10e62bdf422c704'] = 'Итого';
$_MODULE['<{bankrupay}prestashop>payment_return_5ca0b1b910f393ed1f9f6fa99e414255'] = 'Имя владельца счета';
$_MODULE['<{bankrupay}prestashop>payment_return_d717aa33e18263b8405ba00e94353cdc'] = 'Включить следующие данные';
$_MODULE['<{bankrupay}prestashop>payment_return_984482eb9ff11e6310fef641d2268a2a'] = 'Название банка';
$_MODULE['<{bankrupay}prestashop>payment_return_bb4ec5aac6864a05fcc220cccd8e82f9'] = 'Не забудьте указать номер вашего заказа %d в назначении платежа перевода.';
$_MODULE['<{bankrupay}prestashop>payment_return_1faa25b80a8d31e5ef25a78d3336606d'] = 'Не забудьте указать код вашего заказа %s в назначении платежа перевода.';
$_MODULE['<{bankrupay}prestashop>payment_return_19c419a8a4f1cd621853376a930a2e24'] = 'Email с информацией о заказе отправлен на вашу почту.';
$_MODULE['<{bankrupay}prestashop>payment_return_b9a1cae09e5754424e33764777cfcaa0'] = 'Мы отправим ваш заказ сразу по получении оплаты.';
$_MODULE['<{bankrupay}prestashop>payment_return_ca7e41a658753c87973936d7ce2429a8'] = 'По вопросам и за дополнительной информацией, пожалуйста, обращайтесь в';
$_MODULE['<{bankrupay}prestashop>payment_return_66fcf4c223bbf4c7c886d4784e1f62e4'] = 'службу поддержки клиентов';
$_MODULE['<{bankrupay}prestashop>payment_return_d15feee53d81ea16269e54d4784fa123'] = 'Мы обнаружили проблемы с вашим заказом. Если вы считаете, что это ошибка, свяжитесь с нашей';
$_MODULE['<{bankrupay}prestashop>payment_5e1695822fc5af98f6b749ea3cbc9b4c'] = 'Оплата банковским переводом';
$_MODULE['<{bankrupay}prestashop>payment_4e1fb9f4b46556d64db55d50629ee301'] = '(обработка заказа продлится дольше)';
$_MODULE['<{bankrupay}prestashop>payment_return_decce112a9e64363c997b04aa71b7cb8'] = 'клиентской службой.';


return $_MODULE;


controllers/front/payment.php
<?php
/*
* 2018 Antonio Solo
*
*  @author Antonio Solo <as@solotony.com>
*  @copyright  2018 Antonio Solo
*  @license    http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
*/

class BankrupayPaymentModuleFrontController extends ModuleFrontController
{
	public $ssl = true;
	public $display_column_left = false;

	/**
	 * @see FrontController::initContent()
	 */
	public function initContent()
	{
		parent::initContent();

		$cart = $this->context->cart;
		if (!$this->module->checkCurrency($cart))
			Tools::redirect('index.php?controller=order');

		$this->context->smarty->assign(array(
			'nbProducts' => $cart->nbProducts(),
			'cust_currency' => $cart->id_currency,
			'currencies' => $this->module->getCurrency((int)$cart->id_currency),
			'total' => $cart->getOrderTotal(true, Cart::BOTH),
			'this_path' => $this->module->getPathUri(),
			'this_path_bw' => $this->module->getPathUri(),
			'this_path_ssl' => Tools::getShopDomainSsl(true, true).__PS_BASE_URI__.'modules/'.$this->module->name.'/'
		));

		$this->setTemplate('payment_execution.tpl');
	}
}


controllers/front/validation.php
<?php
/*
* 2018 Antonio Solo
*
*  @author Antonio Solo <as@solotony.com>
*  @copyright  2018 Antonio Solo
*  @license    http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
*/

class BankrupayValidationModuleFrontController extends ModuleFrontController
{
	/**
	 * @see FrontController::postProcess()
	 */
	public function postProcess()
	{
		$cart = $this->context->cart;
		if ($cart->id_customer == 0 || $cart->id_address_delivery == 0 || $cart->id_address_invoice == 0 || !$this->module->active)
			Tools::redirect('index.php?controller=order&step=1');

		// Check that this payment option is still available in case the customer changed his address just before the end of the checkout process
		$authorized = false;
		foreach (Module::getPaymentModules() as $module)
			if ($module['name'] == 'bankrupay')
			{
				$authorized = true;
				break;
			}
		if (!$authorized)
			die($this->module->l('This payment method is not available.', 'validation'));

		$customer = new Customer($cart->id_customer);
		if (!Validate::isLoadedObject($customer))
			Tools::redirect('index.php?controller=order&step=1');

		$currency = $this->context->currency;
		$total = (float)$cart->getOrderTotal(true, Cart::BOTH);
		$mailVars = array(
            '{bankrupay_org}' => Configuration::get('BANK_RUPAY_ORG'),
            '{bankrupay_inn}' => nl2br(Configuration::get('BANK_RUPAY_INN')),
            '{bankrupay_kpp}' => nl2br(Configuration::get('BANK_RUPAY_KPP')),
            '{bankrupay_rs}' => nl2br(Configuration::get('BANK_RUPAY_RS')),
            '{bankrupay_bank}' => nl2br(Configuration::get('BANK_RUPAY_BANK')),
            '{bankrupay_bik}' => nl2br(Configuration::get('BANK_RUPAY_BIK')),
            '{bankrupay_ks}' => nl2br(Configuration::get('BANK_RUPAY_KS'))
		);

		$this->module->validateOrder($cart->id, Configuration::get('PS_OS_BANKWIRE'), $total, $this->module->displayName, NULL, $mailVars, (int)$currency->id, false, $customer->secure_key);
		Tools::redirect('index.php?controller=order-confirmation&id_cart='.$cart->id.'&id_module='.$this->module->id.'&id_order='.$this->module->currentOrder.'&key='.$customer->secure_key);
	}
}


bankrupay.php
<?php
/*
* 2018 Antonio Solo
*
*  @author Antonio Solo <as@solotony.com>
*  @copyright  2018 Antonio Solo
*  @license    http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
*/

if (!defined('_PS_VERSION_'))
    exit;

class BankRuPay extends PaymentModule
{
    protected $_html = '';
    protected $_postErrors = array();

    public $details;
    public $owner;
    public $address;
    public $extra_mail_vars;
    public function __construct()
    {
        $this->name = 'bankrupay';
        $this->tab = 'payments_gateways';
        $this->version = '1.0.0';
        $this->author = 'Antonio Solo';
        $this->controllers = array('payment', 'validation');
        $this->is_eu_compatible = 0;

        $this->currencies = true;
        $this->currencies_mode = 'checkbox';

        $config = Configuration::getMultiple(array(
            'BANK_RUPAY_ORG',
            'BANK_RUPAY_INN',
            'BANK_RUPAY_KPP',
            'BANK_RUPAY_RS',
            'BANK_RUPAY_BANK',
            'BANK_RUPAY_BIK',
            'BANK_RUPAY_KS'
        ));
        if (!empty($config['BANK_RUPAY_ORG']))      $this->org = $config['BANK_RUPAY_ORG'];
        if (!empty($config['BANK_RUPAY_INN']))      $this->inn = $config['BANK_RUPAY_INN'];
        if (!empty($config['BANK_RUPAY_KPP']))      $this->kpp = $config['BANK_RUPAY_KPP'];
        if (!empty($config['BANK_RUPAY_RS']))       $this->rs = $config['BANK_RUPAY_RS'];
        if (!empty($config['BANK_RUPAY_BANK']))     $this->bank = $config['BANK_RUPAY_BANK'];
        if (!empty($config['BANK_RUPAY_BIK']))      $this->bik = $config['BANK_RUPAY_BIK'];
        if (!empty($config['BANK_RUPAY_KS']))       $this->ks = $config['BANK_RUPAY_KS'];

        $this->bootstrap = true;
        parent::__construct();

        $this->displayName = $this->l('Безналичный платеж');
        $this->description = $this->l('Прием безналичных платежей на расчетный счет в российском банке');
        $this->confirmUninstall = $this->l('Ты хорошо подумал, друг?');
        $this->ps_versions_compliancy = array('min' => '1.6', 'max' => '1.6.99.99');

        if (!isset($this->org) || !isset($this->inn) || !isset($this->rs) || !isset($this->bank) || !isset($this->bik) || !isset($this->ks))
            $this->warning = $this->l('Надо настроить банковские реквизиты ');
        if (!count(Currency::checkPaymentCurrencies($this->id)))
            $this->warning = $this->l('No currency has been set for this module.');

        $this->extra_mail_vars = array(
            '{bankrupay_org}' => Configuration::get('BANK_RUPAY_ORG'),
            '{bankrupay_inn}' => nl2br(Configuration::get('BANK_RUPAY_INN')),
            '{bankrupay_kpp}' => nl2br(Configuration::get('BANK_RUPAY_KPP')),
            '{bankrupay_rs}' => nl2br(Configuration::get('BANK_RUPAY_RS')),
            '{bankrupay_bank}' => nl2br(Configuration::get('BANK_RUPAY_BANK')),
            '{bankrupay_bik}' => nl2br(Configuration::get('BANK_RUPAY_BIK')),
            '{bankrupay_ks}' => nl2br(Configuration::get('BANK_RUPAY_KS'))
        );
    }

    public function install()
    {
        if (!parent::install() || !$this->registerHook('payment') || ! $this->registerHook('displayPaymentEU') || !$this->registerHook('paymentReturn'))
            return false;
        return true;
    }

    public function uninstall()
    {
        if (!Configuration::deleteByName('BANK_RUPAY_ORG')
            || !Configuration::deleteByName('BANK_RUPAY_INN')
            || !Configuration::deleteByName('BANK_RUPAY_KPP')
            || !Configuration::deleteByName('BANK_RUPAY_RS')
            || !Configuration::deleteByName('BANK_RUPAY_BANK')
            || !Configuration::deleteByName('BANK_RUPAY_BIK')
            || !Configuration::deleteByName('BANK_RUPAY_KS')
            || !parent::uninstall())
            return false;
        return true;
    }

    protected function _postValidation()
    {
        if (Tools::isSubmit('btnSubmit'))
        {
            if (!Tools::getValue('BANK_RUPAY_ORG'))
                $this->_postErrors[] = $this->l('Получатель платежа должен быть указан.');
            elseif (!Tools::getValue('BANK_RUPAY_INN'))
                $this->_postErrors[] = $this->l('ИНН должен быть указан.');
            //elseif (!Tools::getValue('BANK_RUPAY_KPP'))
            //    $this->_postErrors[] = $this->l('КПП должен быть указан.');
            elseif (!Tools::getValue('BANK_RUPAY_RS'))
                $this->_postErrors[] = $this->l('Расчетный счет должен быть указан.');
            elseif (!Tools::getValue('BANK_RUPAY_BANK'))
                $this->_postErrors[] = $this->l('Банк должен быть указан.');
            elseif (!Tools::getValue('BANK_RUPAY_BIK'))
                $this->_postErrors[] = $this->l('БИК должен быть указан.');
            elseif (!Tools::getValue('BANK_RUPAY_KS'))
                $this->_postErrors[] = $this->l('Корреспондентский счет должен быть указан.');
        }
    }

    protected function _postProcess()
    {
        if (Tools::isSubmit('btnSubmit'))
        {
            Configuration::updateValue('BANK_RUPAY_ORG',    Tools::getValue('BANK_RUPAY_ORG'));
            Configuration::updateValue('BANK_RUPAY_INN',    Tools::getValue('BANK_RUPAY_INN'));
            Configuration::updateValue('BANK_RUPAY_KPP',    Tools::getValue('BANK_RUPAY_KPP'));
            Configuration::updateValue('BANK_RUPAY_RS',     Tools::getValue('BANK_RUPAY_RS'));
            Configuration::updateValue('BANK_RUPAY_BANK',   Tools::getValue('BANK_RUPAY_BANK'));
            Configuration::updateValue('BANK_RUPAY_BIK',    Tools::getValue('BANK_RUPAY_BIK'));
            Configuration::updateValue('BANK_RUPAY_KS',     Tools::getValue('BANK_RUPAY_KS'));
        }
        $this->_html .= $this->displayConfirmation($this->l('Settings updated'));
    }

    protected function _displayBankRuPay()
    {
        return $this->display(__FILE__, 'infos.tpl');
    }

    public function getContent()
    {
        if (Tools::isSubmit('btnSubmit'))
        {
            $this->_postValidation();
            if (!count($this->_postErrors))
                $this->_postProcess();
            else
                foreach ($this->_postErrors as $err)
                    $this->_html .= $this->displayError($err);
        }
        else
            $this->_html .= '<br />';

        $this->_html .= $this->_displayBankRuPay();
        $this->_html .= $this->renderForm();

        return $this->_html;
    }

    public function hookPayment($params)
    {
        if (!$this->active)
            return;
        if (!$this->checkCurrency($params['cart']))
            return;

        $this->smarty->assign(array(
            'this_path' => $this->_path,
            'this_path_bw' => $this->_path,
            'this_path_ssl' => Tools::getShopDomainSsl(true, true).__PS_BASE_URI__.'modules/'.$this->name.'/'
        ));
        return $this->display(__FILE__, 'payment.tpl');
    }

    public function hookDisplayPaymentEU($params)
    {
        if (!$this->active)
            return;

        if (!$this->checkCurrency($params['cart']))
            return;

        $payment_options = array(
            'cta_text' => $this->l('Безналичный платеж'),
            'logo' => Media::getMediaPath(_PS_MODULE_DIR_.$this->name.'/bankrupay.jpg'),
            'action' => $this->context->link->getModuleLink($this->name, 'validation', array(), true)
        );

        return $payment_options;
    }

    public function hookPaymentReturn($params)
    {
        if (!$this->active)
            return;

        $state = $params['objOrder']->getCurrentState();
        if (in_array($state, array(Configuration::get('PS_OS_BANKWIRE'), Configuration::get('PS_OS_OUTOFSTOCK'), Configuration::get('PS_OS_OUTOFSTOCK_UNPAID'))))
        {
            $this->smarty->assign(array(
                'total_to_pay' => Tools::displayPrice($params['total_to_pay'], $params['currencyObj'], false),
                'bankrupayOrg' => $this->org,
                'bankrupayInn' => Tools::nl2br($this->inn),
                'bankrupayKpp' => Tools::nl2br($this->kpp),
                'bankrupayRs' => Tools::nl2br($this->rs),
                'bankrupayBank' => Tools::nl2br($this->bank),
                'bankrupayBik' => Tools::nl2br($this->bik),
                'bankrupayKs' => Tools::nl2br($this->ks),
                'status' => 'ok',
                'id_order' => $params['objOrder']->id
            ));
            if (isset($params['objOrder']->reference) && !empty($params['objOrder']->reference))
                $this->smarty->assign('reference', $params['objOrder']->reference);
        }
        else
            $this->smarty->assign('status', 'failed');
        return $this->display(__FILE__, 'payment_return.tpl');
    }

    public function checkCurrency($cart)
    {
        $currency_order = new Currency($cart->id_currency);
        $currencies_module = $this->getCurrency($cart->id_currency);

        if (is_array($currencies_module))
            foreach ($currencies_module as $currency_module)
                if ($currency_order->id == $currency_module['id_currency'])
                    return true;
        return false;
    }

    public function renderForm()
    {
        $fields_form = array(
            'form' => array(
                'legend' => array(
                    'title' => $this->l('Contact details'),
                    'icon' => 'icon-envelope'
                ),
                'input' => array(
                    array(
                        'type' => 'text',
                        'label' => $this->l('Получатель платежа'),
                        'name' => 'BANK_RUPAY_ORG',
                        'required' => true
                    ),
                    array(
                        'type' => 'textarea',
                        'label' => $this->l('ИНН'),
                        'name' => 'BANK_RUPAY_INN',
                        'desc' => $this->l('ИНН 10 или 12 символов'),
                        'required' => true
                    ),
                    array(
                        'type' => 'textarea',
                        'label' => $this->l('КПП'),
                        'name' => 'BANK_RUPAY_KPP',
                        'required' => false
                    ),
                    array(
                        'type' => 'textarea',
                        'label' => $this->l('Расчетный счет'),
                        'name' => 'BANK_RUPAY_RS',
                        'required' => true
                    ),
                    array(
                        'type' => 'textarea',
                        'label' => $this->l('Банк'),
                        'name' => 'BANK_RUPAY_BANK',
                        'required' => true
                    ),
                    array(
                        'type' => 'textarea',
                        'label' => $this->l('БИК'),
                        'name' => 'BANK_RUPAY_BIK',
                        'required' => true
                    ),
                    array(
                        'type' => 'textarea',
                        'label' => $this->l('Корреспондентский счет'),
                        'name' => 'BANK_RUPAY_KS',
                        'required' => true
                    ),
                ),
                'submit' => array(
                    'title' => $this->l('Save'),
                )
            ),
        );

        $helper = new HelperForm();
        $helper->show_toolbar = false;
        $helper->table = $this->table;
        $lang = new Language((int)Configuration::get('PS_LANG_DEFAULT'));
        $helper->default_form_language = $lang->id;
        $helper->allow_employee_form_lang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') ? Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') : 0;
        $this->fields_form = array();
        $helper->id = (int)Tools::getValue('id_carrier');
        $helper->identifier = $this->identifier;
        $helper->submit_action = 'btnSubmit';
        $helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false).'&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name;
        $helper->token = Tools::getAdminTokenLite('AdminModules');
        $helper->tpl_vars = array(
            'fields_value' => $this->getConfigFieldsValues(),
            'languages' => $this->context->controller->getLanguages(),
            'id_language' => $this->context->language->id
        );

        return $helper->generateForm(array($fields_form));
    }

    public function getConfigFieldsValues()
    {
        return array(
            'BANK_RUPAY_ORG'    => Tools::getValue('BANK_RUPAY_ORG',    Configuration::get('BANK_RUPAY_ORG')),
            'BANK_RUPAY_INN'    => Tools::getValue('BANK_RUPAY_INN',    Configuration::get('BANK_RUPAY_INN')),
            'BANK_RUPAY_KPP'    => Tools::getValue('BANK_RUPAY_KPP',    Configuration::get('BANK_RUPAY_KPP')),
            'BANK_RUPAY_RS'     => Tools::getValue('BANK_RUPAY_RS',     Configuration::get('BANK_RUPAY_RS')),
            'BANK_RUPAY_BANK'   => Tools::getValue('BANK_RUPAY_BANK',   Configuration::get('BANK_RUPAY_BANK')),
            'BANK_RUPAY_BIK'    => Tools::getValue('BANK_RUPAY_BIK',    Configuration::get('BANK_RUPAY_BIK')),
            'BANK_RUPAY_KS'     => Tools::getValue('BANK_RUPAY_KS',     Configuration::get('BANK_RUPAY_KS')),
        );
    }
}
